tests/unit/src/test/scala/tests/LogBackupSuite.scala

```diff
@@ -0,0 +1,93 @@
+package tests
+
+import java.nio.file.Files
+
+import scala.util.control.NonFatal
+
+import scala.meta.internal.io.PathIO
+import scala.meta.internal.metals.Directories
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.RecursivelyDelete
+import scala.meta.internal.metals.logging.MetalsLogger
+import scala.meta.internal.metals.utils.LimitedFilesManager
+import scala.meta.io.AbsolutePath
+
+class LogBackupSuite extends BaseSuite {
+  val maxLogBackups = 2
+  val serverConfig: MetalsServerConfig =
+    MetalsServerConfig.default.copy(maxLogFileSize = 100, maxLogBackups = 2)
+  var workspace: AbsolutePath = _
+  def log: AbsolutePath = workspace.resolve(Directories.log)
+  def backupDir: AbsolutePath =
+    workspace.resolve(".metals").resolve(".backup_logs")
```

Comment:
Btw. we should add an option to send the logs automatically and add them to the zip (not default)